test/e2e/dataplane/service_discovery.go

```diff
@@ -119,13 +109,32 @@ func verifyClusterIpWithDig(f *framework.Framework, cluster framework.ClusterInd
 	}, func(result interface{}) (bool, string, error) {
 		doesContain := strings.Contains(result.(string), serviceIP)
 		if doesContain && !shouldContain {
-			return false, fmt.Sprintf("expected execution result %q not to contain %q", result, serviceIP), nil
+			return false || logOnly, fmt.Sprintf("expected execution result %q not to contain %q", result, serviceIP), nil
 		}
 
 		if !doesContain && shouldContain {
-			return false, fmt.Sprintf("expected execution result %q to contain %q", result, serviceIP), nil
+			return false || logOnly, fmt.Sprintf("expected execution result %q to contain %q", result, serviceIP), nil
 		}
+		By(fmt.Sprintf("Validating dig result: %q", result))
```

Comment:
Shouldn't this be before the ifs?

Reply:
Oh right. I refactored my initial fix with logOnly, forgot to move it.